Source/BoundaryConditions/WarpX_PEC.cpp

```diff
@@ -763,187 +879,201 @@ PEC::ApplyReflectiveBoundarytoJfield(
-                                        ? -1._rt : 1._rt;
-                psign[icomp][idim][1] = ( (particle_boundary_hi[idim] == ParticleBoundaryType::Reflecting)
-                                        ||(particle_boundary_hi[idim] == ParticleBoundaryType::Thermal)
-                                        ||(field_boundary_hi[idim] == FieldBoundaryType::PMC))
-                                        ? -1._rt : 1._rt;
-            }
+            amrex::Real pmc_sign = 1._rt;
+            if (!is_tangent_to_bndy) { pmc_sign = -1._rt; }
+
+            // Set psign on lo side
+            psign[idim][icomp][0] = ( (particle_boundary_lo[idim] == ParticleBoundaryType::Reflecting)
+                                  ||  (particle_boundary_lo[idim] == ParticleBoundaryType::Thermal)
+                                  ||  (field_boundary_lo[idim] == FieldBoundaryType::PMC) )
+                                      ? pmc_sign : -pmc_sign;
+
+            // Set psign on hi side
+            psign[idim][icomp][1] = ( (particle_boundary_hi[idim] == ParticleBoundaryType::Reflecting)
+                                  ||  (particle_boundary_hi[idim] == ParticleBoundaryType::Thermal)
+                                  ||  (field_boundary_hi[idim] == FieldBoundaryType::PMC) )
+                                      ? pmc_sign : -pmc_sign;
         }
-        // Set the correct mirror cell calculation for each current
-        // component. Seeing as domain was set to be nodal, nodal fields have
-        // boundaries on domain_lo and domain_hi, but cell-centered fields
-        // have valid cells ranging from domain_lo to domain_hi - 1.
-        mirrorfac[0][idim][0] = 2*domain_lo[idim] - (1 - Jx_nodal[idim]);
-        mirrorfac[0][idim][1] = 2*domain_hi[idim] - (1 - Jx_nodal[idim]);
-        mirrorfac[1][idim][0] = 2*domain_lo[idim] - (1 - Jy_nodal[idim]);
-        mirrorfac[1][idim][1] = 2*domain_hi[idim] - (1 - Jy_nodal[idim]);
-        mirrorfac[2][idim][0] = 2*domain_lo[idim] - (1 - Jz_nodal[idim]);
-        mirrorfac[2][idim][1] = 2*domain_hi[idim] - (1 - Jz_nodal[idim]);
+
+        // Set the mirror index offset on lo and hi sides
+        mirrorfac[idim][0][0] = 2*domain_lo[idim] - (1 - Jx_nodal[idim]);
+        mirrorfac[idim][0][1] = 2*domain_hi[idim] - (1 - Jx_nodal[idim]);
+        mirrorfac[idim][1][0] = 2*domain_lo[idim] - (1 - Jy_nodal[idim]);
+        mirrorfac[idim][1][1] = 2*domain_hi[idim] - (1 - Jy_nodal[idim]);
+        mirrorfac[idim][2][0] = 2*domain_lo[idim] - (1 - Jz_nodal[idim]);
+        mirrorfac[idim][2][1] = 2*domain_hi[idim] - (1 - Jz_nodal[idim]);
+
     }
 
-    // Each current component is handled separately below, starting with Jx.
-    grown_domain_box.convert(Jx_nodal);
 #ifdef AMREX_USE_OMP
 #pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
 #endif
-    for (amrex::MFIter mfi(*Jx); mfi.isValid(); ++mfi) {
+    for (amrex::MFIter mfi(*Jx, false); mfi.isValid(); ++mfi) {
```

Comment:
As a follow-up PR: could you add a comment that explains why you had to add `false` here?